llvm/test/SYCLLowerIR/PropagateAspectsUsage/PropagateFunctionWithSyclDetailAspect/propagate-aspects-from-function-with-sycl-detail-2.ll

```diff
@@ -0,0 +1,96 @@
+; RUN opt -passes=sycl-propagate-aspects-usage %s -S | FileCheck %s
+
+; Check: that baz() takes a mix of "!sycl_used_aspects" of bar() & foo()
+
+;    baz()
+;   /     \
+;  v       v
+; bar()   foo()
+
+source_filename = "main.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%class.anon = type { i8 }
+
+define weak_odr dso_local spir_kernel void @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_() {
+entry:
+  %__SYCLKernel = alloca %class.anon, align 1
+  %__SYCLKernel.ascast = addrspacecast %class.anon* %__SYCLKernel to %class.anon addrspace(4)*
+  call spir_func void @_ZZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_ENKUlvE_clEv(%class.anon addrspace(4)* noundef align 1 dereferenceable_or_null(1) %__SYCLKernel.ascast)
+  ret void
+}
+
+define internal spir_func void @_ZZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_ENKUlvE_clEv(%class.anon addrspace(4)* noundef align 1 dereferenceable_or_null(1) %this) align 2 {
+entry:
+  call spir_func void @_Z3bazv()
+  ret void
+}
```

Comment:
Aspect propagation should work without any kernels, so you can simplify the test here if you are not checking aspect propagation up to the kernel

Reply:
Done [a6da336](https://github.com/intel/llvm/pull/7035/commits/a6da336c7d130956827dc6a7c3c9a6fafd966aad)